### PR TITLE
issue-1469: extended tutor

### PR DESCRIPTION
**Pull request by oliverfoster**
fixes: https://github.com/adaptlearning/adapt_framework/issues/1469

### Added
* `overlay` and `inline` feedback options
* option of a notify text button at the bottom rather than a close button top right
* `_classes` to allow classes to be added to the feedback

### Updated 
* readme https://github.com/adaptlearning/adapt-contrib-tutor/tree/issue/1469#settings
* schema https://github.com/adaptlearning/adapt-contrib-tutor/blob/issue/1469/properties.schema

### CODE REVIEW

**Comment by oliverfoster** [comment]
i need help with schemas on this one. tutor + feedback are quite disjointed.

File: js/tutorOverlay.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
+define([
+    'coreJS/adapt'
```
Comment (brian-learningpool) [inline js/tutorOverlay.js]: `core/js/adapt`?

**Comment by oliverfoster** [comment]
@tomgreenfield is taking this one for investigation

**Comment by oliverfoster** [comment]
@guywillis can you have a look at the styling
@oliverfoster can you do the accessibility testing

**Comment by tomgreenfield** [comment]
See [`questionView.js#L300`](https://github.com/adaptlearning/adapt_framework/blob/b5989a12d36f31f64d5d2e43cfeca189ea5af39f/src/core/js/views/questionView.js#L300) for the line in core where we could do with a `{ deferScroll: true }`, to stop jerking on reset.

**Comment by guywillis** [comment]
Some things to consider:

* `"_type": "none",` - Turns off the feedback but leaves the show feedback button active. Should this be allowed?
* `"_type": "inline",` + `"_hasNotifyBottomButton": false,` - Hide feedback button is still at the bottom of the inline feedback
* `"_type": "overlay",` + `"_hasNotifyBottomButton": true,` - Hide feedback button is still the close icon in the top right of the overlay feedback

As an aside:

* Is anyone aware of a method or config option to force buttons to be full width?

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Comment by oliverfoster** [comment]
> Some things to consider:
> 
> 1. `"_type": "none",` - Turns off the feedback but leaves the show feedback button active. Should this be allowed?
> 2. `"_type": "inline",` + `"_hasNotifyBottomButton": false,` - Hide feedback button is still at the bottom of the inline feedback
> 3. `"_type": "overlay",` + `"_hasNotifyBottomButton": true,` - Hide feedback button is still the close icon in the top right of the overlay feedback
> 
> As an aside:
> 
> 4. Is anyone aware of a method or config option to force buttons to be full width?

1. Yes, as question has `_canShowFeedback` and a different extension may take the feedback for this item... probably? It's an edge case.
2. & 3. Would you prefer `"_hasBottomButton": true,`  ?
4. In what context?

**Comment by oliverfoster** [comment]
@guywillis  could you re-review please? / answer the above questions

File: less/tutor.less [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    .dir-rtl & {
+      right: inherit;
+      left: 0;
+    }
+  }
+
+  &__btn-icon .icon {
+    .icon-cross;
+  }
```
Comment (eleanor-heath) [inline less/tutor.less:21]:
```suggestion
&__btn-icon {
    position: absolute;
    top: 0;
    right: 0;

    .dir-rtl & {
      right: inherit;
      left: 0;
    }
    .icon {
      .icon-cross;
    }
  }
```

**Review by eleanor-heath — APPROVED** [review]

File: less/tutor.less (hunk 2 above)
Comment (oliverfoster) [reply less/tutor.less:21]: @guywillis

**Pull request merged**